packages/http/src/router.ts

```diff
@@ -788,7 +789,8 @@ export class HttpRouter {
                             name: ${JSON.stringify(parameter.parameter.name)},
                             value: parameters.${parameter.parameter.name},
                             query: _query,
-                            parameters: parameters
+                            parameters: parameters,
+                            optional: ${parameter.parameter.isOptional()}
```

Comment:
I think if we want to make type information available in the resolver, its better to make the ReflectionParameter object available there. So 

```typescript
type: ${compiler.reserveVariable('parameterType', parameter.parameter)}
```

 and change the interface `RouteParameterResolverContext` to

```typescript
type: ReflectionParameter;
```

Reply:
fair point!

Reply:
updated!